snapcraft/internal/meta/command.py

```diff
@@ -43,62 +37,134 @@
 )
 
 
-def _get_shebang_from_file(file_path: str) -> List[str]:
+def _get_shebang_from_file(file_path: str) -> Optional[str]:
     """Returns the shebang from file_path."""
-    if not os.path.exists(file_path):
-        raise errors.ShebangNotFoundError()
-
     with open(file_path, "rb") as exefile:
         if exefile.read(2) != b"#!":
-            raise errors.ShebangNotFoundError()
+            return None
         shebang_line = exefile.readline().strip().decode("utf-8")
 
-    # posix is set to False to respect the quoting of variables.
-    shebang_parts = shlex.split(shebang_line, posix=False)
-    # remove the leading /usr/bin/env.
-    if shebang_parts[0] == "/usr/bin/env":
-        shebang_parts = shebang_parts[1:]
-    # or if the shebang startswith /.
-    elif shebang_parts[0].startswith("/"):
-        raise errors.ShebangInRoot()
-
-    return shebang_parts
+    return shebang_line
 
 
-def _find_executable(*, binary: str, prime_dir: str) -> str:
-    found_path: Optional[str] = None
+def _find_executable(*, command: str, prime_dir: str) -> Optional[str]:
     binary_paths = (
-        os.path.join(p, binary) for p in common.get_bin_paths(root=prime_dir)
+        os.path.join(p, command) for p in common.get_bin_paths(root=prime_dir)
     )
     for binary_path in binary_paths:
         if _executable_is_valid(binary_path):
-            found_path = binary_path
-            break
+            return binary_path
```

Comment:
Mostly a stylistic comment, but if you are going to return in every loop, why keep the else?

The code was how it was to keep the single return point, which if with this new approach is too hard to keep then so be it, but the else should not be required anymore.

Reply:
Good point, I saw it and forgot about it, fixed!